repo/fsrepo/config_test.go

```diff
@@ -0,0 +1,217 @@
+package fsrepo
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"reflect"
+	"testing"
+
+	config "github.com/ipfs/go-ipfs/repo/config"
+)
+
+var defaultConfig = []byte(`{
+    "StorageMax": "10GB",
+    "StorageGCWatermark": 90,
+    "GCPeriod": "1h",
+    "Spec": {
+      "mounts": [
+        {
+          "child": {
+            "path": "blocks",
+            "shardFunc": "/repo/flatfs/shard/v1/next-to-last/2",
+            "sync": true,
+            "type": "flatfs"
+          },
+          "mountpoint": "/blocks",
+          "prefix": "flatfs.datastore",
+          "type": "measure"
+        },
+        {
+          "child": {
+            "compression": "none",
+            "path": "datastore",
+            "type": "levelds"
+          },
+          "mountpoint": "/",
+          "prefix": "leveldb.datastore",
+          "type": "measure"
+        }
+      ],
+      "type": "mount"
+    },
+    "HashOnRead": false,
+    "BloomFilterSize": 0
+}`)
+
+var leveldbConfig = []byte(`{
+            "compression": "none",
+            "path": "datastore",
+            "type": "levelds"
+}`)
+
+var flatfsConfig = []byte(`{
+            "path": "blocks",
+            "shardFunc": "/repo/flatfs/shard/v1/next-to-last/2",
+            "sync": true,
+            "type": "flatfs"
+}`)
+
+var measureConfig = []byte(`{
+          "child": {
+            "path": "blocks",
+            "shardFunc": "/repo/flatfs/shard/v1/next-to-last/2",
+            "sync": true,
+            "type": "flatfs"
+          },
+          "mountpoint": "/blocks",
+          "prefix": "flatfs.datastore",
+          "type": "measure"
+}`)
+
+func TestDefaultDatastoreConfig(t *testing.T) {
+	dir, err := ioutil.TempDir("", "ipfs-datastore-config-test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(dir) // clean up
+
+	config := new(config.Datastore)
+	err = json.Unmarshal(defaultConfig, config)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	dsc, err := AnyDatastoreConfig(config.Spec)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	expected := "/blocks:{flatfs;blocks;/repo/flatfs/shard/v1/next-to-last/2};/:{levelds;datastore};"
```

Comment:
yeap, understood about the log and measure bit (would apply to adding caches, and other fun things).

That syntax still feels weird. Since we're going for something the user should be able to read, i would prefer if it was just a little bit more json-ish and used a `[]` for the mounts list.

cc @Kubuxu for thoughts on the syntax